src/Internal.AspNetCore.BuildTools.Tasks/GetGitCommitInfo.cs

```diff
@@ -78,6 +89,13 @@ public override bool Execute()
             return true;
         }
 
+        private bool ResolveFromDetachedHead(string head)
+        {
+            CommitHash = head;
+            Log.LogWarning("The repo in '{0}' appears to be in detached HEAD mode. Unable to determine current git branch.", RepositoryRootPath);
```

Comment:
Yes, we still want this warning when we are unable to resolve the name of the branch being used. But there should be fewer warnings, and they won't cause build to fail. We can make improvements to KoreBuild so that the warning appears only once per solution instead of once per project.